src/main/java/spoon/support/JavaOutputProcessor.java

```diff
@@ -44,19 +46,28 @@
  */
 public class JavaOutputProcessor extends AbstractProcessor<CtNamedElement> implements FileGenerator<CtNamedElement> {
 	PrettyPrinter printer;
-
-	File directory;
+	private Environment environment;
 
 	List<File> printedFiles = new ArrayList<>();
 
+	public JavaOutputProcessor(PrettyPrinter printer) {
+		this.printer = printer;
+		this.environment = printer.getEnvironment();
```

Comment:
we can replace `printer.getEnvironment()` by `getFactory().getEnvironment()`. And then we don't need the new method in interface PrettyPrinter